Make sure we init the memory we allocate (file-4.16 did this, but
file-4.17 doesnt).

http://bugs.gentoo.org/126012

--- file-4.17/src/apprentice.c
+++ file-4.17/src/apprentice.c
@@ -380,8 +380,8 @@ apprentice_file(struct magic_set *ms, st
 		return -1;
 	}
 
-        maxmagic = MAXMAGIS;
-	if ((marray = malloc(maxmagic * sizeof(*marray))) == NULL) {
+	maxmagic = MAXMAGIS;
+	if ((marray = calloc(maxmagic, sizeof(*marray))) == NULL) {
 		(void)fclose(f);
 		file_oomem(ms);
 		return -1;
@@ -509,7 +509,7 @@ parse(struct magic_set *ms, struct magic
 	char *t;
 	private const char *fops = FILE_OPS;
 	uint32_t val;
-	uint32_t cont_level, cont_count;
+	uint32_t cont_level;
 
 	cont_level = 0;
 
